datatables.go

```diff
@@ -302,7 +302,15 @@ func (p *Parser) bindNewPlayer(playerEntity st.IEntity) {
 				}
 				cache[i2] = entityID
 
-				wep := &p.weapons[entityID]
+				wep := p.weapons[entityID]
+
+				if wep == nil {
+					// Something is clearly wrong here
+					// But since we had an empty Equipment instance here before the change
+					// from array with default elements to map, let's create a new instance.
+					wep = new(common.Equipment)
```

Comment:
note that we won't have a `UniqueID` in this case. I know it wasn't there before either, but I'm wondering if we should add it or not. I think this could actually be a common case (or maybe it's indeed never used. unfortunately coecov isn't showing the coverage on this)

Reply:
Actually the only case when I faced this were in test when `OnCreate` were unintentional called on fake player entity because (if I'm not mistaken) its mocked method returned a non-mocked instance of `*st.Property` which have lead to this.

I believe that in reality this can only happen in some corrupted demos when a player entity would have a handle for unexisting entity so in this case we probably should create some error message (though I'm not very familiar with your way of creating an error during the parsing process so if I am the one to implement this behaviour I'd probably need some help). I'm not 100% sure but I've parsed about 10k demos using your parser and never had issues with.

Reply:
Fair point. we can raise a `ParserWarn` event if it ocurrs, like this: https://github.com/markus-wa/demoinfocs-golang/blob/7f5a03d810e41d0d78e8f7c2642e1d69c25957ce/game_events.go#L25

I prefer not aborting if it's not strictly needed, in the past the demo format changed occasionally where previous edge-cases suddenly became the norm, that way we can potentially avoid the need for a hotfix (though I'm pretty sure in this case this won't happen)

Reply:
Right now I'm running the tests and it seems like in the very beggining of some demos there actually are these situations when a player has some weapon which hasn't beed binded prior to binding a player, so I'm having a mismatch with default.expected. What do you think we should do? May be leaving this as-is is actually not a bad idea because we don't really have any information about the weapon aside from its entity ID.

Reply:
Just to clearify, does it mean that we should remove the `ParserWarn` or we should update `default.golden` instead? I suppose that it is the former as I don't see any lines with `ParserWarn` inside `default.expected`.

Reply:
oh, right. Sorry I wrote it to read like we should remove `ParserWarn` but actually now that you say that it makes sense to update `default.golden`.

I thought the diff was the `UniqueID`s, but it's the `ParserWarn`s that are new, which is intended.

so you can run `go test -update` (I think)

Reply:
Alright let's do it. Right now this is the only thing that doesn't allow tests to pass (everything is showing as OK if a remove the lines with `ParserWarn` dispatching) so I do hope that I will not break anything with it.